### DISCUSSION TITLE
Idea: Space-construction zones have a flagger waving a slow sign

### DISCUSSION THREAD

**Discussion by jelster** [jelster/space-truckers#20] [Ideas]
What other type of highway traffic is there? Opportunity for AI- controlled NPC

**Comment by jelster** [comment]
This [article](https://www.sciencedaily.com/releases/2020/12/201209094216.htm) gives a bit of astronomical background on a "solar system super-highway"

**Comment by eldinor** [comment]
I suppose we could have an array of 20-30 text quotes of such type, to educate/entertain the player during scene changing (loading, cutscenes etc).

**Reply by jelster** [reply]
That’s a nice touch too! 

In areas of the route we could have the type of hazard be a construction zone - this could be a nice opportunity to introduce some humor with the character of a construction flag waver and workers. The flagger might also appear in loading screens to vamp with a short animation while things load like you said. Reuse of assets and consistency for players!

**Reply by jelster** [reply]
When I think of what this character’s like, a couple examples come to mind (not necessarily the physical appearance more than the way they behave):

Minions from Despicable Me
CL4P-TR4P from Borderlands
That squirrel from Ice Age

you can count on this character to get into some sort of slap-stick disaster usually of their own making

**Reply by jelster** [reply]
This is a rough sketch I made of a flagger ![image](https://user-images.githubusercontent.com/1508500/106397272-187c1880-63d2-11eb-9a91-aae8b7d0f5d8.png). Wanted to give us something to benchmark against!

**Reply by eldinor** [reply]
So you think it should be animated humanoid model?
How much models in total we'll need?

(One of my client projects was not fulfilled because the total size of all models which he wanted gradually loomed so large - over 500 Mb - so it was impossible have decent render in realtime)

**Reply by jelster** [reply]
I was thinking this could be a sprite, for similar reasons and because I don't think they will need to have a wide range of potential animations, just the flag waving, something funny that happens when the truck hits them (no gore). A static sitting position would let us reuse the images as drivers of construction vehicles or other tbd road traffic.

**Reply by eldinor** [reply]
If they will be 2D (flat) I see no perf problems.

**Comment by jelster** [comment]
Some road signs I dug up from free online sources. I've got a 70MB zip of more where these came from:

![CW11-10](https://user-images.githubusercontent.com/1508500/106371259-034aaf80-6328-11eb-96bc-fe021c711dab.png)
![CW20-7](https://user-images.githubusercontent.com/1508500/106371261-0a71bd80-6328-11eb-94be-169d7bc59824.png)
![W8-22](https://user-images.githubusercontent.com/1508500/106371286-44db5a80-6328-11eb-91bb-c22c56a7505b.png)
